File: Submissions/TriCodeX/ShopEase/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String sku;
  final String category;
  final double price;
  final int quantity;

  Product({
    required this.id,
    required this.name,
    required this.sku,
    required this.category,
    required this.price,
    required this.quantity,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['_id'],
        name: json['name'],
        sku: json['sku'],
        category: json['category'],
        price: json['price'].toDouble(),
        quantity: json['quantity'],
      );

  Map<String, dynamic> toJson() => {
        'name': name,
        'sku': sku,
        'category': category,
        'price': price,
        'quantity': quantity,
      };
}

File: Submissions/TriCodeX/ShopEase/shopease-backend/models/product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  sku: { type: String, required: true },
  category: { type: String },
  price: { type: Number, required: true },
  quantity: { type: Number, default: 0 },
});

// This line avoids the overwrite error:
module.exports = mongoose.models.Product || mongoose.model('Product', productSchema);